test_suite/tests/json.rs

```diff
@@ -360,6 +364,27 @@ fn test_enum() {
     }));
 }
 
+#[test]
+fn test_ranges() {
+    #[derive(TypeInfo)]
+    struct TypeWithRanges {
+        open_range: Range<u8>,
+        closed_range: RangeInclusive<u64>,
+    }
+
+    assert_json_for_type::<TypeWithRanges>(json!({
+        "path": ["json", "TypeWithRanges"],
+        "def": {
+            "composite": {
+                "fields": [
+                    { "name": "open_range", "type": 0, "typeName": "Range<u8>" },
+                    { "name": "closed_range", "type": 2, "typeName": "RangeInclusive<u64>" },
```

Comment:
I think this test should also check that the `inclusive` field is set correctly on the respective types. Might be easiest by just comparing the whole serialized type registry which we do in some other tests.